service/config_provider.go

```diff
@@ -102,42 +136,56 @@ func MustNewConfigProvider(
 // Notice: This API is experimental.
 func NewConfigProvider(
 	locations []string,
-	configMapProviders map[string]configmapprovider.Provider,
-	cfgMapConverters []config.MapConverterFunc,
-	configUnmarshaler configunmarshaler.ConfigUnmarshaler) ConfigProvider {
+	opts ...ConfigProviderOption) (ConfigProvider, error) {
+
+	configOpts := &ConfigProviderOptions{}
+	for _, o := range opts {
+		o(configOpts)
+	}
+
+	if len(locations) == 0 {
+		return nil, fmt.Errorf("cannot create ConfigProvider: no locations provided")
+	}
 	// Safe copy, ensures the slice cannot be changed from the caller.
 	locationsCopy := make([]string, len(locations))
 	copy(locationsCopy, locations)
 	return &configProvider{
 		locations:          locationsCopy,
-		configMapProviders: configMapProviders,
-		cfgMapConverters:   cfgMapConverters,
-		configUnmarshaler:  configUnmarshaler,
+		configMapProviders: configOpts.configMapProviders,
+		cfgMapConverters:   configOpts.cfgMapConverters,
+		configUnmarshaler:  configOpts.configUnmarshaler,
 		watcher:            make(chan error, 1),
-	}
+	}, nil
 }
 
+// Deprecated: use NewDefaultConfigProvider instead
 // MustNewDefaultConfigProvider returns the default ConfigProvider, and it creates configuration from a file
 // defined by the given configFile and overwrites fields using properties.
 func MustNewDefaultConfigProvider(configLocations []string, properties []string) ConfigProvider {
-	return NewDefaultConfigProvider(configLocations, properties)
+	configProvider, err := NewDefaultConfigProvider(configLocations, properties)
+	if err != nil {
+		panic(err)
+	}
+	return configProvider
 }
 
-// Deprecated: [v.0.44.0] use MustNewDefaultConfigProvider instead
 // NewDefaultConfigProvider returns the default ConfigProvider, and it creates configuration from a file
 // defined by the given configFile and overwrites fields using properties.
-func NewDefaultConfigProvider(configLocations []string, properties []string) ConfigProvider {
+func NewDefaultConfigProvider(configLocations []string, properties []string) (ConfigProvider, error) {
```

Comment:
This should become the default values for `NewConfigProvider`. So I would remove this and change NewConfigProvider to have these map providers (user will be able to override this with the `WithCfgMapProviders`) and these map converters (user will be able to override this with the `WithCfgMapConverters`).